lms/envs/common.py

```diff
@@ -589,6 +591,12 @@
 COURSE_ID_PATTERN = COURSE_KEY_PATTERN.replace('course_key_string', 'course_id')
 COURSE_KEY_REGEX = COURSE_KEY_PATTERN.replace('P<course_key_string>', ':')
 
+# Pattern to match a course key or a library key
+COURSELIKE_KEY_PATTERN = r'(?P<course_key_string>({}|{}))'.format(
+    r'[^/:+]+/[^/:+]+/[^/:+]+',
+    r'[^/:]+:[^/+]+\+[^/+]+(\+[^/]+)?',
+)
```

Comment:
Yes, taking some commits from before I rebased & squashed, the original move is here: https://github.com/mitocw/edx-platform/commit/b6413e8def359123d851da0e6e2146133159d1c7

And the change to the regex itself is here: https://github.com/mitocw/edx-platform/commit/8c80b47996b70c2d8c9be365c873045a1065c1ff

Note the explanation:
> This fixes some instances of incorrect matching of org/course/run
> keys. For example:
> "library-v1:MITx+TL/import_status/library.3HtOcx.tar.gz" was being
> matched as a course key, even though the valid key is
> "library-v1:MITx+TL" in this case.
>
> Studio doesn't let users use `+` or `:` symbols when filling out the
> org/course/run for new courses, so I just factored that into the
> org/course/run pattern.